cantools/db/attribute_definition.py

```diff
@@ -0,0 +1,98 @@
+class AttributeDefinition(object):
+    """A definition of an attribute that can be associated with attributes in nodes/messages/signals
+
+    """
+
+    def __init__(self,
+                 name,
+                 defaultValue=None,
+                 kind=None,
+                 type_=None,
+                 minimum=None,
+                 maximum=None,
+                 choices=None):
+        self._name = name
+        self._defaultValue = defaultValue
+        self._kind=kind
+        self._type_ = type_
+        self._minimum=minimum
+        self._maximum=maximum
+        self._choices = choices
+
+    @property
+    def name(self):
+        """The attribute name as a string.
+
+        """
+
+        return self._name
+
+    @property
+    def defaultValue(self):
+        """The default value that this attribute has, or ``None`` if unavailable.
+
+        """
+
+        return self._defaultValue
+
+    @defaultValue.setter
+    def defaultValue(self, value):
+        self._defaultValue = value
+
+    @property
+    def kind(self):
+        """The attribute kind (BU_, BO_, SG_), or ``None`` if unavailable.
+
+        """
+
+        return self._kind
+
+    @property
+    def type_(self):

```

Comment:
I think I prefer `type_name` over `type_`, as ending a name with underscore in the interface looks weird. Internally I'm fine with ending variables with underscore if the name is a Python keyword.